src/main/java/com/coveo/FMT.java

```diff
@@ -1,210 +1,28 @@
-        formatSourceFilesInDirectory(file);
-      } else {
-        formatSourceFile(file);
-      }
-    }
-  }
-
-  private FileFilter getFileFilter() {
-    if (verbose) {
-      logger.debug("Filter files on '" + filesNamePattern + "'.");
-    }
-    return new FileFilter() {
-      @Override
-      public boolean accept(File pathname) {
-        return pathname.isDirectory() || pathname.getName().matches(filesNamePattern);
-      }
-    };
-  }
-
-  private void formatSourceFile(File file) {
-    if (file.isDirectory()) {
-      logger.info("File '" + file + "' is a directory. Skipping.");
-      return;
-    }
-
-    if (verbose) {
-      logger.debug("Formatting '" + file + "'.");
-    }
-
-    CharSource source = Files.asCharSource(file, Charsets.UTF_8);
-    CharSink sink = Files.asCharSink(file, Charsets.UTF_8);
-    try {
-      String input = source.read();
-      String formatted = formatter.formatSource(input);
-      formatted = RemoveUnusedImports.removeUnusedImports(formatted, JavadocOnlyImports.KEEP);
-      formatted = ImportOrderer.reorderImports(formatted);
-      if (!input.equals(formatted)) {
-        if (!validateOnly) {
-          sink.write(formatted);
-        }
-        nonComplyingFiles += 1;
-      }
-      filesFormatted.add(file.getAbsolutePath());
-      if (filesFormatted.size() % 100 == 0) {
-        logNumberOfFilesFormatted();
-      }
-    } catch (FormatterException e) {
-      logger.warn("Failed to format file '" + file + "'.", e);
-    } catch (IOException e) {
-      logger.warn("Failed to format file '" + file + "'.", e);
-    }
-  }
-
-  private void handleMissingDirectory(String directoryDisplayName, File directory)
-      throws MojoFailureException {
-    if (failOnUnknownFolder) {
-      String message =
-          directoryDisplayName
-              + " directory '"
-              + directory
-              + "' does not exist, failing build (failOnUnknownFolder is true).";
-      logger.error(message);
-      throw new MojoFailureException(message);
-    } else {
-      logger.warn(
-          directoryDisplayName + " directory '" + directory + "' does not exist, ignoring.");
-    }
-  }
-
-  private void logNumberOfFilesFormatted() {
-    logger.info(
-        String.format(
-            "Processed %d files (%d %s).",
-            filesFormatted.size(),
-            nonComplyingFiles,
-            (validateOnly ? "non-complying" : "reformatted")));
-  }
+  @Deprecated
+  @Parameter(defaultValue = "false", property = "validateOnly")
+  private boolean validateOnly;
 
-  private void maybeFailIfNonComplying() throws MojoFailureException {
-    if (validateOnly && nonComplyingFiles > 0) {
-      String message =
-          "Found "
-              + nonComplyingFiles
-              + " non-complying files, failing build (validateOnly is true)";
-      logger.error(message);
-      throw new MojoFailureException(message);
-    }
+  @Override
+  @SuppressWarnings("deprecation")
```

Comment:
Is that not enough on the field? (which is what's annotated with `@Parameter`)

(or is Maven so broken/unintuitive that it'll take the javadoc from the accessor?)